ggshield/core/git_hooks/ci/get_scan_ci_parameters.py

```diff
@@ -0,0 +1,108 @@
+import os
+from pathlib import Path
+from typing import Dict, Optional, Tuple, Union
+
+import click
+
+from ggshield.core.errors import NotAMergeRequestError, UnexpectedError
+from ggshield.utils.git_shell import get_commits_not_in_branch, get_remotes
+
+from .supported_ci import SupportedCI
+
+
+def travis_scan_ci_args() -> Tuple[str, str]:
+    if "TRAVIS_PULL_REQUEST" not in os.environ:
+        raise NotAMergeRequestError()
+    try:
+        commit_range = os.environ["TRAVIS_COMMIT_RANGE"]
+        first_commit, last_commit = commit_range.split("..")
+        return last_commit, first_commit + "~1"
+    except Exception as exc:
+        raise UnexpectedError("Failed to extract scan ci arguments for travis") from exc
+
+
+# Note: this does not exist (yet ?) for CircleCI, see
+# https://circleci.canny.io/config/p/provide-env-variable-for-branch-name-targeted-by-pull-request
+CI_TARGET_BRANCH_ASSOC: Dict[SupportedCI, str] = {
+    SupportedCI.GITHUB: "GITHUB_BASE_REF",
+    SupportedCI.GITLAB: "CI_MERGE_REQUEST_TARGET_BRANCH_NAME",
+    SupportedCI.JENKINS: "CHANGE_TARGET",
+    SupportedCI.AZURE: "SYSTEM_PULLREQUEST_TARGETBRANCHNAME",
+    SupportedCI.BITBUCKET: "BITBUCKET_PR_DESTINATION_BRANCH",
+    SupportedCI.DRONE: "DRONE_COMMIT_BRANCH",
+}
+
+
+def get_scan_ci_parameters(
+    current_ci: SupportedCI,
+    wd: Optional[Union[str, Path]] = None,
+    verbose: bool = False,
+) -> Union[Tuple[str, str], None]:
+    """
+    Function used to gather current commit and reference commit, for the SCA/IaC scan
+    ci commands.
+    Return: None if the MR has no associated commits, otherwise a tuple (current_commit, reference commit)
+
+    Note: this function will not work if the git directory is a shallow clone
+    """
+    if verbose:
+        click.echo(
+            f"\tIdentified current ci as {current_ci.value}",
+            err=True,
+        )
+
+    if current_ci == SupportedCI.TRAVIS:
+        return travis_scan_ci_args()
+
+    remotes = get_remotes(wd=wd)
+    if len(remotes) == 0:
+        if verbose:
+            click.echo(
+                "\tNo remote found.",
+                err=True,
+            )
+        remote_prefix = ""
+    else:
+        if verbose:
+            click.echo(
+                f"\tUsing first remote {remotes[0]}.",
+                err=True,
+            )
+        remote_prefix = f"{remotes[0]}/"
+
+    target_branch_var = CI_TARGET_BRANCH_ASSOC.get(current_ci)
+    if not target_branch_var:
+        raise UnexpectedError(f"Using scan ci is not supported for {current_ci.value}.")
```

Comment:
nit: Should this be done before the `get_remotes` so that in the case of CircleCI we directly raise this error?

Reply:
I think it's ok, getting the remotes is a very light operation (does not involve network) - and also the case where there is no remote should not happen in practice, it was mostly used for tests (although that's not best practice :see_no_evil: )

Reply:
Note that we do have one test where a remote is set though